docs/technical_reference/flowsheets/generic_desalination_train.rst

```diff
@@ -0,0 +1,108 @@
+    * Desalination unit cost
+        * Using base treatment cost:
+            * m.fs.{unit_name}.desalter.base_cost (in $/m^3 of product)
+        * Defining cost as a linear function of cost per water recovery and offset which will compute the base case as base_cost=recovery_cost*water_recovery+offset
+            * m.fs.{unit_name}.desalter.brine_solids_concentration (kg/m^3) 
+            * m.fs.{unit_name}.desalter.brine_water_mass_percent (%) 
+    * Desalination unit operation (Fix only one)
+        * Water recovery (m.fs.{unit_name}.desalter.water_recovery (in %))
+        * (or) Brine solids concentration (m.fs.{unit_name}.desalter.water_recovery (in %))
+        * (or) Brine water content (m.fs.{unit_name}.desalter.water_recovery (in %))
+    * Valorization unit costs: 
+        * Base cost (m.fs.Pretreatment.costing.base_cost in $/m^3 of feed) 
+        * (Optional) Target product removal value (m.fs.Valorizer.separator.product_value[{component}] in % (Default: 0 for all components))
+        * (Optional) Chemical cost (m.fs.Valorizer.separator.chemical_cost in $/kg (Default: 0)
+    * Valorization unit operation: 
+        * (Optional) Target product removal amount (m.fs.Valorizer.separator.component_removal_percent[{component] in % (Default: 0 for all components))
+        * (Optional) Chemical addition amount (m.fs.Valorizer.separator.additive_cost in PPM if any (Default: 0) 
+
+
+The flowsheet is populated with baseline costs for the unit operations based on general literature values, as presented in Figure 1. 
+
+
+Flowsheet Specifications
+------------------------
+
+These are the default values configured in flowsheet main function: 
+
+.. csv-table::
+   :header: "Description", "Value", "Units"
+
+   "**Feed Water**"
+   "Water mass flow","1", ":math:`\text{kg/s}`"
+   "TDS concentration", "3.5", ":math:`\text{g/L}`"
+   "ion X", "0.3", ":math:`\text{g/L}`"
+   "Temperature", "298.15", ":math:`\text{K}`"
+   "Pressure", "101325", ":math:`\text{Pa}`"
+
+   "**Stream costing**"
+   "Product distribution", "0.0", ":math:`\text{$/m^3}`"
+   "Feed sourcing", "0.0", ":math:`\text{$/m^3}`"
+   "Waste disposal", "1.0", ":math:`\text{$/m^3}`"
+
+   "**Pretreatment**"
+   "Base cost", "0.0", ":math:`\text{$/m^3}`"
+   "Additive cost", "0.0", ":math:`\text{$/kg}`"
+   "Additive dose", "1.0", ":math:`\text{PPM}`"
+   "Component removal percent [X]", "50", ":math:`\text{%}`"
+   "Separation cost [X]", "0.5", ":math:`\text{$/kg}`"
+
+   "**Desal_1**"
+   "Base cost", "0.3", ":math:`\text{$/m^3}`"
+   "Recovery", "80", ":math:`\text{%}`"
+
+   "**Desal_2**"
+   "Base cost", "0.5", ":math:`\text{$/m^3}`"
+   "Recovery cost", "0.01", ":math:`\text{$/m^3/%wr}`"
+   "Recovery cost offset", "35", ":math:`\text{%}`"
+   "Recovery", "50", ":math:`\text{%}`"
+
+   "**Desal_3**"
+   "Base cost", "10", ":math:`\text{$/m^3}`"
+   "Brine water mass percent", "80", ":math:`\text{%}`"
```

Comment:
I'm not sure why these units aren't displaying.... everything appears to be right to me, but ig I'm missing something
![image](https://github.com/watertap-org/watertap/assets/96305519/b6228d30-009a-4248-b11e-1ad4689c8d23)

Reply:
I think I know now. The exponents aren't handled correctly - they should be outside of the brackets. See some of the following examples
![image](https://github.com/watertap-org/watertap/assets/96305519/a88f9e6e-328b-4b78-b78a-a57d744fe07d)

Reply:
Odd worked fine in vscode rst viewer (I hate formatting rst files - it works perfectly in vs code rst viewer but here always glitches out on watertap git)

I updated the rst formating to what you suggested. 

![image](https://github.com/watertap-org/watertap/assets/33663878/83933492-125d-45a3-b719-ee1bff5c1ef7)

Reply:
Hmmm, I'm thoroughly confused now... this issue is still unresolved. I'm not gonna have time to look into it until Monday probably. The CodeCov has also been failing and the problematic files aren't even being touched in this PR. Any thoughts on this documentation issue and the CodeCov issue @lbianchi-lbl (feel free to respond on Monday)

Reply:
@MarcusHolly I've looked into it, and I think the drop in coverage might be due to the fact that 3 of the entry points in `setup.py` have been changed to point to a `watertap.flowsheet.case_studies` subpackage, but the corresponding `.py` files have not been moved:

![2024-06-29_21-30](https://github.com/watertap-org/watertap/assets/48035537/84370643-9e8d-4e60-833c-763ebf08c46e)
![2024-06-29_21-29](https://github.com/watertap-org/watertap/assets/48035537/34d39222-6ad8-4def-982b-7684e2b33897)

If the entry points point to nonexisting modules, then the corresponding `watertap.ui.fsapi.test_flowsheet_interfaces` tests would not run due to `ModuleNotFoundError`s being raised  when trying to import those modules, but the failures would be silent/hidden because of #1384.

If this is the case, it should be enough to revert the changes to L141:143 of `setup.py` (or actually move the files into a new `case_studies` subfolder to match the updated entry points, although I don't recall this being discussed as a change we planned to make before the release).

Reply:
Thanks Ludovico!

Reply:
The only catch here - is those flowsheets are not in a case_studies folder? - at least not in current pr state?

Reply:
Yea, I've fixed the setup.py file, but #1438 did a lot of restructuring, specifically cleaning up all the extraneous folders we had in the flowsheet directory. So now instead of watertap.examples.flowsheets.case_studies... , all of the flowsheets have their respective folders under watertap.flowsheets.

Reply:
Yeah, I miss read the comment and though we should still have a case_studies folder, but that is not the case (e.g. flowsheet.case_studies) but that was me miss reading the comment.  Thanks for all the help with this PR!